gql/resolver_lid.go

```diff
@@ -0,0 +1,63 @@
+package gql
+
+import (
+	"context"
+
+	gqltypes "github.com/filecoin-project/boost/gql/types"
+)
+
+type dealData struct {
+	Indexed         gqltypes.Uint64
+	FlaggedUnsealed gqltypes.Uint64
+	FlaggedSealed   gqltypes.Uint64
+}
+
+type pieces struct {
+	Indexed         int32
+	FlaggedUnsealed int32
+	FlaggedSealed   int32
+}
+
+type sectorUnsealedCopies struct {
+	Unsealed int32
+	Sealed   int32
+}
+
+type sectorProvingState struct {
+	Active   int32
+	Inactive int32
+}
+
+type lidState struct {
+	DealData             dealData
+	Pieces               pieces
+	SectorUnsealedCopies sectorUnsealedCopies
+	SectorProvingState   sectorProvingState
+	FlaggedPieces        int32
+}
+
+// query: lid: [LID]
+func (r *resolver) LID(ctx context.Context) (*lidState, error) {
```

Comment:
I'd suggest creating a different resolver method for each of these subsections (one for DealData, one for Pieces etc).

That way each one is loaded separately, and if one of them fails or is slow it doesn't affect the others.

Reply:
These should not fail, as all the data is coming from Boost and should be cached by that point - I'd rather we make 1 API call for a few int.